aws-ecs/45-iam-role.tf

```diff
@@ -0,0 +1,151 @@
+resource "aws_iam_role" "pvault_ecs" {
+  name = "pvault-ecs-role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = "sts:AssumeRole"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+      },
+    ]
+  })
+}
+
+data "aws_arn" "db_password" {
+  arn = aws_secretsmanager_secret.db_password.arn
+}
+
+resource "aws_iam_policy" "pvault_secrets" {
+  name        = "pvault-secrets"
+  description = "pvault-secrets policy"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "secretsmanager:GetSecretValue",
+        ]
+        Resource = [
+          "arn:${data.aws_arn.db_password.partition}:${data.aws_arn.db_password.service}:${data.aws_arn.db_password.region}:${data.aws_arn.db_password.account}:secret:/pvault/*"
+        ]
+      },
+    ]
+  })
+}
+
+resource "aws_iam_policy" "pvault_kms" {
+  name        = "pvault-kms"
+  description = "pvault-kms policy"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "kms:*",
```

Comment:
Narrow it down to `kms:Encrypt` and `kms:Decrypt`

Reply:
Great idea! Done